### ISSUE TITLE
Outgoing Messages No Longer Sent

### ISSUE THREAD

**Issue by cwmke** [kfatehi/matrix-puppet-imessage#38]
A month or so ago, the power went out at my house. After rebooting I can no longer send messages from clients. However, incoming messages still work and messages sent from Apple's messages app are imported properly. Any ideas on resolving this are greatly appreciated.

MacOS: 10.13.6

```
Error in handleMatrixEvent { Error: could not determine third party room id!
    at App._handleMatrixMessageEvent (/Users/<USER>/matrix-puppet-imessage/node_modules/matrix-puppet-bridge/src/base.js:1034:13)
    at App.handleMatrixMessageEvent (/Users/<USER>/matrix-puppet-imessage/node_modules/matrix-puppet-bridge/src/base.js:1015:25)
    at App.handleMatrixEvent (/Users/<USER>/matrix-puppet-imessage/node_modules/matrix-puppet-bridge/src/base.js:1007:19)
    at Bridge._onConsume (/Users/<USER>/matrix-puppet-imessage/node_modules/matrix-appservice-bridge/lib/bridge.js:856:26)
    at tryCatcher (/Users/<USER>/matrix-puppet-imessage/node_modules/matrix-appservice-bridge/node_modules/bluebird/js/main/util.js:26:23)
    at Promise.successAdapter (/Users/<USER>/matrix-puppet-imessage/node_modules/matrix-appservice-bridge/node_modules/bluebird/js/main/nodeify.js:23:30)
    at Promise._settlePromiseAt (/Users/<USER>/matrix-puppet-imessage/node_modules/matrix-appservice-bridge/node_modules/bluebird/js/main/promise.js:582:21)
    at Promise._settlePromises (/Users/<USER>/matrix-puppet-imessage/node_modules/matrix-appservice-bridge/node_modules/bluebird/js/main/promise.js:700:14)
    at Async._drainQueue (/Users/<USER>/matrix-puppet-imessage/node_modules/matrix-appservice-bridge/node_modules/bluebird/js/main/async.js:123:16)
    at Async._drainQueues (/Users/<USER>/matrix-puppet-imessage/node_modules/matrix-appservice-bridge/node_modules/bluebird/js/main/async.js:133:10)
    at Immediate.Async.drainQueues (/Users/<USER>/matrix-puppet-imessage/node_modules/matrix-appservice-bridge/node_modules/bluebird/js/main/async.js:15:14)
    at runCallback (timers.js:705:18)
    at tryOnImmediate (timers.js:676:5)
    at processImmediate (timers.js:658:5)
  [stack]:
   'Error: could not determine third party room id!\n    at App._handleMatrixMessageEvent (/Users/<USER>/matrix-puppet-imessage/node_modules/matrix-puppet-bridge/src/base.js:1034:13)\n    at App.handleMatrixMessageEvent (/Users/<USER>/matrix-puppet-imessage/node_modules/matrix-puppet-bridge/src/base.js:1015:25)\n    at App.handleMatrixEvent (/Users/<USER>/matrix-puppet-imessage/node_modules/matrix-puppet-bridge/src/base.js:1007:19)\n    at Bridge._onConsume (/Users/<USER>/matrix-puppet-imessage/node_modules/matrix-appservice-bridge/lib/bridge.js:856:26)\n    at tryCatcher (/Users/<USER>/matrix-puppet-imessage/node_modules/matrix-appservice-bridge/node_modules/bluebird/js/main/util.js:26:23)\n    at Promise.successAdapter (/Users/<USER>/matrix-puppet-imessage/node_modules/matrix-appservice-bridge/node_modules/bluebird/js/main/nodeify.js:23:30)\n    at Promise._settlePromiseAt (/Users/<USER>/matrix-puppet-imessage/node_modules/matrix-appservice-bridge/node_modules/bluebird/js/main/promise.js:582:21)\n    at Promise._settlePromises (/Users/<USER>/matrix-puppet-imessage/node_modules/matrix-appservice-bridge/node_modules/bluebird/js/main/promise.js:700:14)\n    at Async._drainQueue (/Users/<USER>/matrix-puppet-imessage/node_modules/matrix-appservice-bridge/node_modules/bluebird/js/main/async.js:123:16)\n    at Async._drainQueues (/Users/<USER>/matrix-puppet-imessage/node_modules/matrix-appservice-bridge/node_modules/bluebird/js/main/async.js:133:10)\n    at Immediate.Async.drainQueues (/Users/<USER>/matrix-puppet-imessage/node_modules/matrix-appservice-bridge/node_modules/bluebird/js/main/async.js:15:14)\n    at runCallback (timers.js:705:18)\n    at tryOnImmediate (timers.js:676:5)\n    at processImmediate (timers.js:658:5)',
  [message]: 'could not determine third party room id!' } { age: 118,
  content: { body: 'Test', msgtype: 'm.text' },
  event_id: '$randomstringrandomh_a-eventidearandomstring',
  origin_server_ts: 0000000000000, <number changed>
  room_id: '!roomidchangedyah:<DOMAIN>
  sender: '@<USER>:<DOMAIN>',
  type: 'm.room.message',
  unsigned: { age: 118 },
  user_id: '@<USER>:<DOMAIN>' } ﻿

**Comment by nr23730** [comment]
Yeah, I recently discovered the same issue when restarting one of my bridges. I already pushed a fix to the puppet-bridge base (5cc9b80b623bdb8870df74af4706138235b0c3d5). 

Just needed to apply it to this bridge.

**Comment by cwmke** [comment]
Wow was that fast! Thanks a lot.
